Sources/SwiftDocC/Model/Rendering/Variants/VariantPatchOperation.swift

```diff
@@ -56,26 +56,87 @@ public enum VariantPatchOperation<Value: Codable> {
     }
 }
 
-extension VariantCollection.Variant where Value: RangeReplaceableCollection {
-    /// Applies the variant's patch operations to a given value and returns the patched value.
-    ///
-    /// - Parameter originalValue: The value that the variant will apply the patch operations to.
-    /// - Returns: The value after applying all patch operations.
-    func applyingPatchTo(_ originalValue: Value) -> Value {
-        var result = originalValue
-        for operation in patch {
-            switch operation {
-            case .replace(let newValue):
-                result = newValue
-            case .add(let newValue):
-                result.append(contentsOf: newValue)
-            case .remove:
-                result.removeAll()
+// The synthesized implementation is sufficient for this conformance.
+extension VariantPatchOperation: Equatable where Value: Equatable {}
+
+// MARK: Applying patches
+
+/// A type that can be transformed by incrementally applying variant patch operations.
+protocol VariantCollectionPatchable {
+    /// Apply an "add" patch operation to the value
+    mutating func add(_ other: Self)
+    /// Apply a "remove" patch operation to the value
+    mutating func remove()
+}
+
+extension Optional: VariantCollectionPatchable where Wrapped: VariantCollectionPatchable {
+    mutating func add(_ other: Wrapped?) {
+        guard var wrapped, let other else { return }
```

Comment:
Is `wrapped` referring to the type of the optional? In this case does it has to be unwrapped?

Reply:
`wrapped` is the optional's value (which is of  the type `Wrapped`)